src/Bicep.LangServer/Snippets/SnippetsProvider.cs

```diff
@@ -119,96 +123,65 @@ private IEnumerable<Snippet> GetRequiredPropertiesSnippetsForDisciminatedObjectT
         }
     }
 
-    private Snippet? GetRequiredPropertiesSnippet(ObjectType objectType, string label, string? discriminatedObjectKey = null)
+    private static ObjectSyntax GetObjectSnippetSyntax(ObjectType objectType, ref int tabStopIndex, string? discriminatedObjectKey)
     {
-        int index = 1;
-        StringBuilder sb = new();
-
-        var sortedProperties = objectType.Properties.OrderBy(x =>
+        var typeProperties = objectType.Properties.Values.OrderBy(x =>
+            PropertiesSortPreferenceList.IndexOf(x.Name) switch {
+                -1 => int.MaxValue,
+                int index =>  index,
+            })
+            .Where(TypeHelper.IsRequired);
+
+        var objectProperties = new List<ObjectPropertySyntax>();
+        foreach (var typeProperty in typeProperties)
         {
-            var index = propertiesSortPreferenceList.IndexOf(x.Key);
+            objectProperties.Add(GetObjectPropertySnippetSyntax(typeProperty, ref tabStopIndex, discriminatedObjectKey));
+        }
```

Comment:
Can this be simplified using a `.Select()` call?

Reply:
It felt too fragile - the `tabStopIndex` param needs to be incremented in the correct order, otherwise we could possibly output tab stops in the wrong order - e.g. we could end up with:
```
{
  foo: $1
  nested: {
    bar: $3
  }
  baz: $2
}
```

Excerpt from the LSP spec to give context on tab stops:
<img width="643" alt="image" src="https://github.com/user-attachments/assets/e532863d-49be-4ec0-a443-bbb63db43316">

I'll add a comment to explain this. We have a test that'll get broken if someone tries to refactor this.